### ISSUE TITLE
Testing Setup

### ISSUE THREAD

**Issue by curran** [pachamaltese/d3po#22]
Before I start modifying code towards #16 , I want to be able to verify that everything still works at each step. I see there is a `/dev` directory with several examples in `.html` files. However, it's not clear how to run these. One way I can see is to copy-paste the code from those into the `index.html` of the `src` directory.

What I'm thinking of doing as a next step is to set up a page that shows all those examples at once, zoomed out, so that I can verify with little effort that nothing breaks when I change code. It's not exactly unit tests, but it's the same in spirit (let's call it "visual unit tests").

**Comment by pachadotdev** [comment]
@curran Hi. This is how I tested the html files in dev/

```
cd ~/github
git clone git@github.com:pachamaltese/d3po.git
cd src_d3_35
npm install
npm run build
cp ~/github/d3po/src_d3_35/d3po.js ~/github/d3po/dev/d3po.js
cp ~/github/d3po/inst/htmlwidgets/lib/d3.min.js ~/github/d3po/dev/d3.min.js
cd .. && cd dev
python -m SimpleHTTPServer
```

donut.html, halfdonut.html and treemap.html work well
network.html, by some reason, "forgot" to draw the links in the network.

**Comment by pachadotdev** [comment]
@curran I just updated the previous comment with the actual way to test the visualization

**Comment by curran** [comment]
@pachamaltese Thank you for that! It's a sensible workflow for testing the visualizaion examples one at a time, but what about testing them all at once? I'm imagining flipping through many tabs and refreshing each every time the code is changed. I can start with this but have a feeling the verification workflow is not as efficient as it could be.

**Comment by pachadotdev** [comment]
@curran yes, the actual verification is really bad!

**Comment by curran** [comment]
I followed the above instructions and the results were as follows:

 * `donut.html` OK
 * `geo.html` first attempt
   * 404 not found for `topojson.min.js` and `d3po.full.js`
 * `geo.html` second attempt (after populating `topojson.min.js` and `d3po.full.js`)
   * "Geo Map visualizations require setting the "coords" method."
   *  The `.coords` method is indeed invoked in the example, so not sure what is going on here.
 * `halfdonut.html` OK
 * `network.html` OK
 * `pie.html` OK
 * `scatterplot.html`
   * `Uncaught ReferenceError: size is not defined`
 * `treemap.html` OK

@pachamaltese Could you please investigate the errors in `geo.html` and `scatterplot.html`? Thank you.

**Comment by pachadotdev** [comment]
@curran Hi! Thanks for reporting this!!
In june the scatterplot was working. Now it is "ignoring" the size argument, at least I made the example more explicit.
For the geomap i added a very old coords file I found in an old external drive, but the visualization "ignores" the coords argument.

**Comment by curran** [comment]
@pachamaltese Thanks for the updates! I see you made a commit recently https://github.com/pachamaltese/d3po/commit/9c4f757891e067fbf3f5950bfa93ade6ac3835e4 , but even after those changes, both files are still failing with the same errors on my machine. Did you verify that the changes made the programs work?

**Comment by pachadotdev** [comment]
@curran yes, no cigar. I tried to make it work and it didn't. I'd put these
two on hold and proceed with the others.

**Comment by curran** [comment]
@pachamaltese Ok. I will proceed with the refactoring work even with these two not working.

**Comment by curran** [comment]
Closed in #24
